cmd/ndm_daemonset/probe/deletehandler.go

```diff
@@ -47,15 +47,22 @@ func (pe *ProbeEvent) deleteBlockDevice(bd blockdevice.BlockDevice, bdAPIList *a
 	uuid, ok := generateUUID(bd)
 	// this can happen if the device didn't have any unique identifiers
 	if !ok {
-		klog.Info("could not recreate UUID while removing device")
-		return nil
+		klog.Info("could not recreate GPT-UUID while removing device")
+	}
+
+	// try to deactivate resource with both UUIDs. This is required in the following case
+
+	existingLegacyBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, bd.UUID)
+	if existingLegacyBlockDeviceResource != nil {
+		pe.Controller.DeactivateBlockDevice(*existingLegacyBlockDeviceResource)
+		klog.V(4).Infof("deactivated device: %s, using legacy UUID", bd.DevPath)
 	}
 
 	existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, uuid)
```

Comment:
we should do this is uuid is valid (ok is true).

Reply:
No, for legacy case, the device will have the UUID already generated and stored in the blockDevice UUID field. It is being set by the udev package.